moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java

```diff
@@ -564,6 +562,7 @@ public String nextName() throws IOException {
       result = nextQuotedValue(SINGLE_QUOTE_OR_SLASH);
     } else if (p == PEEKED_BUFFERED_NAME) {
       result = peekedString;
+      peekedString = null;
```

Comment:
unrelated bug fix. No visible effects, but it was wrong to retain this value when it wasn’t reachable.